src/main/java/project/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,38 @@
+package project.dao;
+
+import java.util.List;
+import java.util.Optional;
+import project.db.Storage;
+import project.lib.Dao;
+import project.models.Manufacturer;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return Storage.manufacturers.stream()
+                 .filter(m -> m.getId().equals(id))
+                 .findFirst();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        return Storage.manufacturers.set(Storage.manufacturers.indexOf(manufacturer), manufacturer);
```

Comment:
```suggestion
        IntStream.range(0, Storage.manufacturers.size())
                .filter(i -> Storage.manufacturers.get(i).getId().equals(manufacturer.getId()))
                .forEach(i -> Storage.manufacturers.set(i, manufacturer));
        return manufacturer;
```
In the storage there is no equal model to the input one, you should look for model with the same id only